### DISCUSSION TITLE
Import relationships (postgres)

### DISCUSSION THREAD

**Discussion by mikevarela** [Budibase/budibase#9271] [Help] [answered]
Hi,

Loving the possibilities with this tool. I've created an ERD and later database in Postgres using PGAdmin. I've created all the relationships in that tool and in the resulting SQL statement. I've imported the data source into Budibase but some tables failed, specifically the union (join) many-to-many tables. 

This brings up a point that's hard to find an answer for, can I import relationships built into the database tables already? I see in the docs that Budibase requires fk fields to be built elsewhere (postgres) so why not import the relationships with those? And to this secondary point, how to import union tables?

Thanks so much, really looking forward to using this tool!!

**Comment by melohagan** [comment, marked as answer]
Hey @mikevarela 

Thanks for exploring Budibase!

Currently the tables must have a primary key before they can be imported into Budibase, including joining tables. There is an issue that seems related to this here: https://github.com/Budibase/budibase/issues/4604

Being able to fetch relationships automatically would be useful, although if it's guessing, it may not make the right choice so there'd still be some degree of editing the relationship definitions in some cases: https://github.com/Budibase/budibase/issues/9287

**Reply by mikevarela** [reply]
Thanks, that helps. I've created id's in the union tables and marked them as primary keys. I keep getting errors though when trying to now build relationships in Budibase, specifically with these union tables. I've been able to import all other tables and develop relationships with no issues.

Say for table

talent
------------
id bigserial primary key not null
...

project
------------
id bigserial primary key not null
...


talent_project
------------
id bigserial primary key not null
talent_id
project_id




In the MANY relationship builder, I keep getting the error when trying to mark the 'through' table. The error is

**Ensure all columns in table are nullable or auto generated**

What could be the issue here. I've removed the NOT NULL constraint from the two join columns (talent_id, project_id) of the join table but I'm still getting this error.

**Reply by melohagan** [reply]
Are your id fields auto-generated? If they're not null, then they must be auto-generated to work in Budibase.

**Reply by mikevarela** [reply]
A combination of this finally worked. I had some 'updated' and 'created' columns and set them to NOT NULL and that caused the error. The id field was PK and BIGSERIAL so it was fine.

Thanks
